homeassistant/components/mqtt/climate.py

```diff
@@ -472,6 +519,11 @@ def _setup_from_config(self, config: ConfigType) -> None:
         ):
             support |= ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
 
+        if (self._topic[CONF_HUMIDITY_STATE_TOPIC] is not None) or (
+            self._topic[CONF_HUMIDITY_COMMAND_TOPIC] is not None
+        ):
```

Comment:
```suggestion
        if self._topic[CONF_HUMIDITY_COMMAND_TOPIC] is not None:
```
We can also set the target humidity if  there is no state topic. In that case it will be set in optimistic mode.